qiskit_nature/problems/second_quantization/lattice/models/ising_model.py

```diff
@@ -0,0 +1,143 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Ising model"""
+from typing import Optional
+
+import numpy as np
+from retworkx import PyGraph
+
+from qiskit_nature.operators.second_quantization import SpinOp
+from qiskit_nature.problems.second_quantization.lattice.lattices import Lattice
+
+from .lattice_model import LatticeModel
+
+
+class IsingModel(LatticeModel):
+    """The Ising model."""
+
+    @classmethod
+    def uniform_parameters(
+        cls,
+        lattice: Lattice,
+        uniform_hopping: complex,
+        uniform_onsite_potential: complex,
+        onsite_interaction: complex = None,
+    ) -> "IsingModel":
+        """Set a uniform hopping parameter and on-site potential over the input lattice.
+
+        Args:
+            lattice: Lattice on which the model is defined.
+            uniform_hopping: The hopping parameter. (or uniform_interaction)
+            uniform_onsite_potential: The on-site potential. (or uniform_external_field)
+            onsite_interaction: The strength of the on-site interaction, the Ising model does not
+                have an on-site interaction.
+
+        Returns:
+            The Ising model with uniform parameters.
+
+        Raises:
+            Warning: If the on-site interaction is not None.
+        """
+        if onsite_interaction is not None:
+            raise Warning(
+                "The Ising model does not have on-site interactions. Provided onsite-interaction "
+                "parameter will be ignored."
+            )
+        graph = lattice.graph
+        for node_a, node_b, _ in graph.weighted_edge_list():
+            if node_a != node_b:
+                graph.update_edge(node_a, node_b, uniform_hopping)
+
+        for node_a in graph.node_indexes():
+            if graph.has_edge(node_a, node_a):
+                graph.update_edge(node_a, node_a, uniform_onsite_potential)
+            else:
+                graph.add_edge(node_a, node_a, uniform_onsite_potential)
+
+        return cls(Lattice(graph))
```

Comment:
This part is also entirely identical to the `FermiHubbardModel`. I suggest we move `FermiHubbardModel.uniform_parameters` to `LatticeModel` and here we simply do:
```suggestion
        super().uniform_parameters(lattice, uniform_hopping, uniform_onsite_potential, None)
```